iceoryx_hoofs/vocabulary/include/iox/detail/string_type_traits.hpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_VOCABULARY_STRING_TYPE_TRAITS_HPP
+#define IOX_HOOFS_VOCABULARY_STRING_TYPE_TRAITS_HPP
+
+#include <cstdint>
+
+#include "iceoryx_hoofs/cxx/type_traits.hpp"
+
+namespace iox
+{
+template <uint64_t Capacity>
+class string;
+
+/// @brief struct to check whether an argument is a cxx::string
+template <typename T>
+struct is_cxx_string : std::false_type
+{
+};
+
+template <uint64_t N>
+struct is_cxx_string<::iox::string<N>> : std::true_type
+{
+};
+
+//////////////////
+/// BEGIN TypeInfo
+//////////////////
+
+/// @brief Provides a translation from a type into its human readable name
+/// NOLINTJUSTIFICATION The name should be stored in a compile time variable. Access is always
+///   safe since it is null terminated and always constant. Other alternatives are not available
+///   at compile time.
+/// NOLINTBEGIN(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
+
+template <typename T>
+struct TypeInfo
```

Comment:
I'm a bit torn here. I understand that you want to separate the string type traits from others so that a user doesn't depend on the string when not necessary. But it doesn't feel right to have the `TypeInfo` for the integer types here as well.

Reply:
@FerdinandSpitzschnueffler I understand what you mean. However, I came to this decision after looking at the usage of the header `string_type_traits.hpp`:

1. `functional_interface.inl`
2. `string.hpp`

`TypeInfo` is sort of related to string as well as it `Provides a translation from a type into its human readable name`. Happy to rename the file or split it up into even smaller chunks. But I suppose that's something for a follow PR as this one is already quite big.

Reply:
Fine for now, I guess. I don't see an elegant way to separate `TypeInfo` for integer types and string at the moment.